torch/distributed/launch.py

```diff
@@ -212,6 +212,33 @@ def parse_args():
     parser.add_argument('training_script_args', nargs=REMAINDER)
     return parser.parse_args()
 
+from random import randint
+import socket
+def is_port_in_use(master_address, master_port):
+    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
+        return s.connect_ex((master_address, master_port)) == 0
```

Comment:
Shouldn't we try `bind()` here instead of connect? Wouldn't the connect always fail since nothing is listening on that port yet?

Reply:
Did you mean to say "always succeed"

We are just testing if the port is free here, and unfortunately I can't see how to avoid the race condition here.

Unless I misunderstood your comment, @pritamdamania87